client/types/startsettings.go

```diff
@@ -17,6 +17,10 @@ type StartSettings struct {
 	// Optional additional HTTP headers to send with all HTTP requests.
 	Header http.Header
 
+	// Optional function that can be used to modify the HTTP headers
+	// before each HTTP request.
+	HeaderFunc func(http.Header) http.Header
```

Comment:
Can we further clarify the expectations? Since we always clone before passing the header parameter, I assume it is safe to modify and return the argument, no need to clone it again, right? Perhaps say something like:

// Can modify and return the argument or return the argument without modifying.

Reply:
Yeah, that's correct, it's safe to modify. Updated the comment.